authorizer/src/utils/auth_utils.py

```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/env python
+# -*- coding: utf-8; -*-
+
+# Copyright (c) 2023 Oracle and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/
+import os
+from enum import Enum
+from typing import Dict, List
+
+from oci.auth.signers import (
+    InstancePrincipalsSecurityTokenSigner,
+    get_resource_principals_signer
+)
+from oci.identity_data_plane.models import (
+    AuthenticateClientDetails,
+    AuthenticateClientResult,
+    FilterGroupMembershipDetails,
+    FilterGroupMembershipResult
+)
+from oci.identity_data_plane.models.principal import Principal
+
+from authorizer.src.utils.identity_utils import (
+    AuthenticationException,
+    AuthorizationException,
+    ExtendedIdentityDataPlaneClient
+)
+
+ALLOWED_GROUP_IDS = "GROUP_IDS"
+
+
+class SignerType(Enum):
+    AUTO = 0
+    RESOURCE_PRINCIPAL = 1
+    INSTANCE_PRINCIPAL = 2
+
+
+def get_signer(signer_type: SignerType = SignerType.AUTO):
+    """Gets the corresponding signer from signer type.
+
+    Parameters
+    ----------
+    signer_type: integer
+        The signer type
+
+    Returns
+    -------
+    Signer:
+        An instance of Signer.
+    """
+    if signer_type == SignerType.AUTO:
+        if _get_env_bool("RP_AUTH", False):
+            signer_type = SignerType.RESOURCE_PRINCIPAL
+        else:
+            signer_type = SignerType.INSTANCE_PRINCIPAL
+    if signer_type == SignerType.RESOURCE_PRINCIPAL:
+        return get_resource_principals_signer()
+    else:
+        return _get_internal_instance_principal_signer()
+
+
+def do_authn(
+    identity_client: ExtendedIdentityDataPlaneClient,
+    headers: Dict[str, List[str]]
+) -> Principal:
+    """Performs the authn validation from given headers.
+
+    Parameters
+    ----------
+    identity_client: ExtendedIdentityDataPlaneClient
+        An instance of ExtendedIdentityDataPlaneClient
+    headers: dict
+        A dict of headers to be authenticated
+
+    Returns
+    -------
+    Principal:
+        An instance of Principal
+    """
+    client_details = AuthenticateClientDetails()
+    client_details.request_headers = headers
+    authenticate_response = identity_client.authenticate_client(authenticate_client_details=client_details)
```

Comment:
No, it will raise error, but this error will be catched in the `authorizer.py` class: https://github.com/oracle/oci-mlflow/pull/20/files#diff-1db0dbd6026961d8b8d099510d11f00aed88bf6dbf382704aac4460657c897b4R80. And since we're trying to guess what `(request-target)` is in this example, when we get the exception, it could be just the one that's caused by the wrong guess. So we choose to `pass` instead of logging the error message.